tests/Unit/ControlSystem/Test_Averager.cpp

```diff
@@ -0,0 +1,212 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <cmath>
+#include <cstddef>
+
+#include "ControlSystem/Averager.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.Averager.Linear",
+                  "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.1;
+  const constexpr size_t deriv_order = 2;
+  const double final_time = 5.0;
+
+  // test true and false `using_average_0th_deriv_of_q`
+  Averager<deriv_order> averager_t(0.5, true);
+  Averager<deriv_order> averager_f(0.5, false);
+
+  CHECK(averager_t.using_average_0th_deriv_of_q());
+  CHECK_FALSE(averager_f.using_average_0th_deriv_of_q());
+
+  while (t < final_time) {
+    // test using an analytic function f(t) = t
+    const DataVector analytic_func = {t, 1.0, 0.0};
+
+    // update exponential averager
+    averager_t.update(t, {analytic_func[0]}, {0.1});
+    averager_f.update(t, {analytic_func[0]}, {0.1});
+    // compare values once averager has sufficient data
+    if (averager_t.have_valid_data(t)) {
+      auto result_t = averager_t(t);
+      auto result_f = averager_f(t);
+      // check function value, which should agree with the effective time
+      // in the `average_0th_deriv_of_q`=`true` case and with the true time `t`
+      // in the `false` case
+      CHECK(approx(result_t[0][0]) == averager_t.average_time(t)[0]);
+      CHECK(approx(result_f[0][0]) == t);
+      // check first derivative (same value for both checks)
+      CHECK(approx(result_t[1][0]) == analytic_func[1]);
+      CHECK(approx(result_f[1][0]) == analytic_func[1]);
+      // check second derivative
+      // The exponential averager uses finite differencing to approximate the
+      // derivatives. The second derivative is only a first order approximation,
+      // which is why we enforce a less stringent check here.
+      CHECK(fabs(result_t[2][0] - analytic_func[2]) <= 1.0e-12);
```

Comment:
Use an `Approx` with a custom tolerance: `Approx custom_approx = Approx::custom().epsilon(1.e-12);` or so.  (Unless something is specifically guaranteeing an absolute tolerance of 1e-12, in which case the current version is correct.)